### PR TITLE
docs: improve "why" section

### PR DESCRIPTION
**Pull request by codejet**
Just a few small improvements/corrections to the "Why" section of the docs.

### CODE REVIEW

File: docs/guide/why.md [hunk 1]
```diff
@@ -14,17 +14,17 @@ We recommend you to read the blog post [Reimagine Atomic CSS](https://antfu.me/p
...
 
-UnoCSS inherits Windi's on-demand natural, [attributify mode](/presets/attributify), [shortcuts](/config/shortcuts), [variant groups](/transformers/variant-group), [compilation mode](/transformers/compile-class) and a lot more. On top of that, UnoCSS is built ground up with the maximum extensibility and performance in mind, making us able to introduce new features like [pure CSS icons](/presets/icons), [valueless attributify](/presets/attributify#valueless-attributify), [tagify](/presets/tagify), [web fonts](/presets/web-fonts), etc.
+UnoCSS inherits Windi's on-demand, natural [attributify mode](/presets/attributify), [shortcuts](/config/shortcuts), [variant groups](/transformers/variant-group), [compilation mode](/transformers/compile-class) and a lot more. On top of that, UnoCSS is built ground up with the maximum extensibility and performance in mind, making us able to introduce new features like [pure CSS icons](/presets/icons), [valueless attributify](/presets/attributify#valueless-attributify), [tagify](/presets/tagify), [web fonts](/presets/web-fonts), etc.
```
Comment (antfu) [inline docs/guide/why.md:17]: This was intentional. `on-demand` serves as an adj here. Any better way of expressing that?

Comment (codejet) [reply docs/guide/why.md:17]: But as an adjective of what exactly? "Natural" is an adjective itself, hence to me it looked like those two adjectives apply to Windi's attribute mode.

Comment (antfu) [reply docs/guide/why.md:17]: Oh, good point, then it should be `on-demand nature` 👍

Comment (codejet) [reply docs/guide/why.md:17]: Ah, got it ☺️ Will change it.

**Pull request merged**